File: scala/src/io/stpettersen/blackjack/SoundEffects.scala
/**
 * Blackjack
 * Copyright 2015 Sam Saint-Pettersen
 * Released under the MIT/X11 License.
 *
 * Scala Swing/AWT implementation.
*/

package io.stpettersen.blackjack
import java.util.{List => JList, ArrayList => JArrayList}
import java.io.IOException
import org.newdawn.easyogg.OggClip

/**
 * SoundEffects implements playing game sound effects.
 */
object SoundEffects extends ISoundEffects {

  private val strEffects: Array[String] = Array("deal", "shuffle", "hit", "reveal")
  private val effects: JList[OggClip] = new JArrayList[OggClip]()
  private var soundOn: Boolean = true

  def init(): Unit = {
    var i: Int = 0
    for(i <- 0 to strEffects.length - 1) {
      try {
        effects.add((null)) //new OggClip(String.format("sounds/%s.ogg", strEffects(i))))
      }
      catch {
        case ioe: IOException => ioe.printStackTrace()
      }
    }
  }

  /**
   * Toggle sound effects on/off.
   * @return Are sound effects on?
   */
  def toggle: Boolean = {
    if(soundOn) soundOn = false
    else soundOn = true
    soundOn
  }

  /**
   * Play a sound effect.
   * @param effect Name of sound effect to play.
   */
  def play(effect: String): Unit = {
    if(soundOn) {
      //effects.get(strEffects.indexOf(effect)).play()
    }
  }
}

File: java/io/stpettersen/blackjack/SoundEffects.java
/**
 * Blackjack
 * Copyright 2015 Sam Saint-Pettersen
 * Released under the MIT/X11 License.
 * 
 * Java Swing/AWT implementation.
*/

package io.stpettersen.blackjack;
import java.util.Arrays;
import java.util.List;
import java.util.ArrayList;
import java.io.IOException;
import org.newdawn.easyogg.OggClip;

public class SoundEffects
{
    private static List<String> strEffects;
    private static List<OggClip> effects;
    private static boolean soundOn;
    
    /**
     * SoundEffects implements playing game sound effects.
    */
    public static void init()
    {
        String s = "sounds/";
        String[] names = { "deal", "shuffle", "hit", "reveal" };
        strEffects = Arrays.asList(names);
        try
        {
            effects = new ArrayList<OggClip>();
            for(int i = 0; i < strEffects.size(); i++)
            {
                effects.add(new OggClip(String.format("%s/%s.ogg", s, strEffects.get(i))));
            }
            soundOn = true;
        }
        catch(IOException e)
        {
            e.printStackTrace();
        }
    }
    
    /**
     * Toggle sound effects on/off.
     * @return Are sound effects on?
    */
    public static boolean toggle()
    {
        if(soundOn) soundOn = false;
        else soundOn = true;
        return soundOn;
    }
    
    /**
     * Play a sound effect.
     * @param effect Name of sound effect to play.
    */
    public static void play(String effect)
    {
        if(soundOn)
        {
            effects.get(strEffects.indexOf(effect)).play();
        }
    }
}
